### ISSUE TITLE
STP import error

### ISSUE THREAD

**Issue by agerrius** [assimp/assimp#1183]
I'm trying to read a STP file with assimp, but in C# I get a CTD in my application (no error). I tried as well with the command line tool but that just reports that the file could not be read and in the debugger I can't figure out what is going on.

I have attached the STP file I got from the user. I'm not very familiar with the format, so it might also be that it is not the same version that assimp supports for reading.

[hangar-USCG.zip](https://github.com/assimp/assimp/files/791336/hangar-USCG.zip)

**Comment by kimkulling** [comment]
STP == stepfile?

**Comment by agerrius** [comment]
Yes, it should be a STEP file (made with CATIA). And looking at the content (like the magic number) it seems to be OK.

**Comment by kimkulling** [comment]
When using the wrong extension ( from assimp point of view in this case ) assimp will interpret it as an obj-file and the import will fail :-).

Is STP the correct extension for step-files?

**Comment by agerrius** [comment]
I think STP is the right extension for the format, but I'll double check in the code.

**Comment by kimkulling** [comment]
At the moment these kinds of STEP-files are not supported by the loader. I am currently investigationg what is missing.

**Comment by PaddiM** [comment]
Hi,

STEP and IFC use the same syntax (P21). I know assimp could read IFC files but nothing about STEP. IFCLoader uses STEP reader classes and generated express data (see IFCReaderGen.h). But I haven't seen infos about express data for different STEP AP's.

**Comment by agerrius** [comment]
That might be my misunderstanding then, I saw support for IFC-STEP in the list, but was not aware they are two file formats.

**Comment by kimkulling** [comment]
Could anyone provide a spec for the STEP-file, which is good to understand?

**Comment by PaddiM** [comment]
Official specs are in ISO 10303 (https://en.wikipedia.org/w/index.php?title=ISO_10303).  
But you could more infos on WikiSTEP (http://wikistep.org/index.php/Main_Page) and STEPcode (http://stepcode.org/docs/home/). STEPcode also has EXPRESS schemas (http://stepcode.org/docs/included_schemas/)

**Comment by doug-matthews** [comment]
I was wondering if the STP file has been implemented yet. I saw that it was supported on the README, but I tried using the library to convert a STP file and it wasn't able to open the file properly.

**Comment by asmaloney** [comment]
I tried some STEP files from here:

https://www.cax-if.org/library/index.html

And couldn't get any of the ones I tried to load - a variety of syntax errors.

Given the, uh... variety... of standards I'm not 100% sure these are the kind of STEP files that should be readable by the assimp code?

**Comment by Rayfoundry** [comment]
If a dependency to [OCE](https://github.com/tpaviot/oce) (OpenCascade Community Edition) is no problem, there's a first class CAD kernel based on C++ Assimp could use to read STEP and IGES.

**Comment by makc** [comment]
bump :) what's the stp status? I couldn't open any stp files. here's one: 
[servo_model_nadim.stp.zip](https://github.com/assimp/assimp/files/2687090/servo_model_nadim.stp.zip)

**Comment by kimkulling** [comment]
Currently work in progress. I have to fight with the point-definition stored in the EXPRESS-schema.

**Comment by dibenedetto** [comment]
> If a dependency to [OCE](https://github.com/tpaviot/oce) (OpenCascade Community Edition) is no problem, there's a first class CAD kernel based on C++ Assimp could use to read STEP and IGES.

@Rayfoundry can you please direct to the CAD C++ class in OCE ?

**Comment by VictorVakoms** [comment]
Any news about "step" format? Will it be supported in the next release?

**Comment by fire** [comment]
BRLCAD recommended:

> AP203 (v1 and v2) is by far the most common, followed by AP214.  Both are pretty well supported among commercial CAD at this point.  AP242 is the newest that fixes many problems in 203 and 214, but not fully adopted everywhere yet.

> Stepcode as the library supporting the cad system

**Comment by visusys** [comment]
Any update on more broad STEP file support? Still getting `ERROR: Failed to load file: IFC: Unrecognized file schema: AUTOMOTIVE_DESIGN`

**Comment by maidenone** [comment]
This might be handy, an assembly of 2 simple parts exported with SolidEdge in AP203, AP214 and AP242
[2Shapes.zip](https://github.com/assimp/assimp/files/8669713/2Shapes.zip)

**Comment by rainduprince** [comment]
I Still getting ERROR: Failed to load file: IFC: Unrecognized file schema:????   
assimp 5.2.5

**Comment by Vipul-M-Rawal** [comment]
Any update on .stp file import ?

**Comment by tellypresence** [comment]
Both files attached to this ticket fail to load with "Unrecognized file schema" errors:
[`hangar-USCG.stp`](https://github.com/assimp/assimp/files/791336/hangar-USCG.zip): `IFC: Unrecognized file schema: AUTOMOTIVE DESIGN { 1 0 10303 214 1 1 1 1 }`
[`servo_model_nadim.stp`](https://github.com/assimp/assimp/files/2687090/servo_model_nadim.stp.zip): `IFC: Unrecognized file schema: CONFIG_CONTROL_DESIGN`
